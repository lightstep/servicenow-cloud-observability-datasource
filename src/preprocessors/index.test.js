import { preprocessData } from './index';

test('preprocesses logs successfully', () => {
  const logsDataFrame = preprocessData(
    {
      data: {
        attributes: {
          logs: [
            [
              1691409972788,
              {
                event: 'one',
                severity: 'ErrorSeverity',
                tags: {
                  'http.status_code': 200,
                  large_batch: true,
                  trace_id: 'd29a3fa8fb446ec65eb691a3259a541e',
                },
              },
            ],
            [
              1691409971908,
              {
                event: 'two',
                severity: 'InfoSeverity',
                tags: {
                  customer: 'hipcore',
                  large_batch: false,
                  trace_id: 'd0fa420269652931236c94bc54d2233e',
                },
                k8s_environment: 'production',
                k8s_namespace: 'default',
                k8s_pod: 'hipcore-pod',
              },
            ],
          ],
        },
      },
    },
    { refId: 'a' },
    ''
  );

  expect(logsDataFrame.toJSON()).toMatchSnapshot();
});
