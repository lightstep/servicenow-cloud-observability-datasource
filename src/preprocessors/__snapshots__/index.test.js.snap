// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preprocesses logs successfully 1`] = `
Object {
  "fields": Array [
    Object {
      "config": Object {},
      "labels": undefined,
      "name": "time",
      "type": "time",
      "values": Array [
        1691409972788,
        1691409971908,
      ],
    },
    Object {
      "config": Object {},
      "labels": undefined,
      "name": "content",
      "type": "string",
      "values": Array [
        "one",
        "two",
      ],
    },
    Object {
      "config": Object {},
      "labels": undefined,
      "name": "level",
      "type": "string",
      "values": Array [
        "error",
        "info",
      ],
    },
    Object {
      "config": Object {},
      "labels": undefined,
      "name": "severity",
      "type": "string",
      "values": Array [
        "ErrorSeverity",
        "InfoSeverity",
      ],
    },
    Object {
      "config": Object {},
      "labels": undefined,
      "name": "tags",
      "type": "other",
      "values": Array [
        "{\\"http.status_code\\":200,\\"large_batch\\":true,\\"trace_id\\":\\"d29a3fa8fb446ec65eb691a3259a541e\\"}",
        "{\\"customer\\":\\"hipcore\\",\\"large_batch\\":false,\\"trace_id\\":\\"d0fa420269652931236c94bc54d2233e\\"}",
      ],
    },
    Object {
      "config": Object {},
      "labels": undefined,
      "name": "k8s_environment",
      "type": "string",
      "values": Array [
        undefined,
        "production",
      ],
    },
    Object {
      "config": Object {},
      "labels": undefined,
      "name": "k8s_namespace",
      "type": "string",
      "values": Array [
        undefined,
        "default",
      ],
    },
    Object {
      "config": Object {},
      "labels": undefined,
      "name": "k8s_pod",
      "type": "string",
      "values": Array [
        undefined,
        "hipcore-pod",
      ],
    },
  ],
  "meta": Object {
    "preferredVisualisationType": "logs",
  },
  "name": undefined,
  "refId": "a",
}
`;
